### PR TITLE
[LLVMGPU] Add good tile size for large matmul in tensorcore pipeline

### PR DESCRIPTION
**Pull request by ThomasRaoux**
Based on early analysis we think this is a good tile size for large cases for f16.

### CODE REVIEW

File: compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp [hunk 1]
```diff
@@ -83,30 +84,42 @@ static void getMatmulConfig(SmallVectorImpl<TileWorkgroupSizePair> &tileSizes) {
...
-  tileSizes.push_back(TileWorkgroupSizePair({{32, 128, 32}, {32, 8, 1}}));
-  tileSizes.push_back(TileWorkgroupSizePair({{128, 64, 8}, {16, 8, 1}}));
-  tileSizes.push_back(TileWorkgroupSizePair({{16, 256, 32}, {64, 2, 1}}));
-  tileSizes.push_back(TileWorkgroupSizePair({{8, 32, 32}, {8, 8, 1}}));
-
-  tileSizes.push_back(TileWorkgroupSizePair({{32, 128, 4}, {32, 8, 1}}));
-  tileSizes.push_back(TileWorkgroupSizePair({{8, 128, 4}, {32, 1, 1}}));
-  tileSizes.push_back(TileWorkgroupSizePair({{16, 64, 4}, {16, 2, 1}}));
-  tileSizes.push_back(TileWorkgroupSizePair({{1, 128, 8}, {32, 1, 1}}));
+  tileSizes.push_back(TileWorkgroupSizePair({{32, 128, 32}, {32, 8, 1}, 1}));
```
Comment (manishucsd) [inline compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp:87]: Pipeline depth for all the configurations is set to 1?

File: compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp [hunk 2]
```diff
@@ -83,30 +84,42 @@ static void getMatmulConfig(SmallVectorImpl<TileWorkgroupSizePair> &tileSizes) {
...
+    int64_t parallelDim = M * N;
+    static constexpr int64_t kLargDimThreashold = 1536;
+    // Based on early analysis we found that 128x256x32_3 gives acceptable
+    // performance across many of the large matrix sizes for f16. This needs to
+    // be refined into a better startegy based on empircal data but this gives
+    // us a quick solution to achieve performance in the right order of
+    // magnitude for large square like cases.
+    if (parallelDim >= kLargDimThreashold * kLargDimThreashold) {
+      tileSizes.push_back(
+          TileWorkgroupSizePair({{128, 256, 32}, {128, 2, 1}, 3}));
```
Comment (manishucsd) [inline compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp:115]: I see a better pipeline depth number here.

File: compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp (hunk 1 above)
Comment (ThomasRaoux) [inline compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp:87]: Yes this unchanged compared to before, just moving the place where we set it.

File: compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp (hunk 2 above)
Comment (ThomasRaoux) [reply compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp:115]: what do you mean?

File: compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp (hunk 1 above)
Comment (manishucsd) [reply compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp:87]: Where these used? I think the tile size we intend to use are below.

**Review by manishucsd — APPROVED** [review]

File: compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp (hunk 2 above)
Comment (manishucsd) [reply compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp:115]: I saw `1` for a few pipeline depth field and not sure where is pipeline depth of 1 applicable. A pipeline depth number of 3 or more what I have seen for A100 multi-staged pipeline.

Comment (ThomasRaoux) [reply compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp:115]: the 1 is for the SIMT case where we don't use async.cp

**Pull request merged**